packages/systemd/systemd.spec

```diff
@@ -130,6 +131,15 @@ Summary: Files for resolved
 %description resolved
 %{summary}.
 
+%package unified-cgroup-hierarchy
+Summary: Bootconfig snippet for unified cgroup hierarchy
+Requires: (%{_cross_os}image-feature(unified-cgroup-hierarchy) and %{name})
+Supplements: (%{_cross_os}image-feature(unified-cgroup-hierarchy) and %{name})
```

Comment:
I haven't run across `Supplements` in the wild much, is this a nice to have or do we need this weak backward dependency for this to work?

Reply:
> I haven't run across `Supplements` in the wild much, is this a nice to have or do we need this weak backward dependency for this to work?

The main drawback of weak dependencies - which otherwise have the nice property of affinitizing a package install to certain features - is that the package can be dropped under [certain conditions](https://rpm-software-management.github.io/rpm/manual/dependencies.html):
> On the other hand, if a package that is named foo or that Provides: foo is selected, and a package bar containing “Supplements: foo” exists, then bar is also selected as long as doing so does not lead to unresolvable dependencies.

I can work around this with strong forward dependencies with a bit of effort.